hoomd/box.py

```diff
@@ -0,0 +1,402 @@
+                    [0,  Ly,      Lz * yz],
+                    [0,  0,       Lz]]
+
+
+        Returns:
+            box (hoomd.Box): The created box.
+        """
+        b = cls(0, 0)
+        b.matrix = box_matrix
+        return b
+
+    @classmethod
+    def _from_cpp(self, cpp_obj):
+        b = Box()
+        b._cpp_obj = cpp_obj
+        return b
+
+    # Dimension based properties
+    @property
+    def dimensions(self):
+        """int: The dimensionality of the box.
+
+        If ``Lz == 0``, the box is treated as 2D, otherwise it is 3D. This
+        property is not settable.
+        """
+        return 2 if self.is2D else 3
+
+    @property
+    def is2D(self):
+        """bool: Flag whether the box is 2D.
+
+        If ``Lz == 0``, the box is treated as 2D, otherwise it is 3D. This
+        property is not settable.
+        """
+        return self.Lz == 0
+
+    # Length based properties
+    @property
+    def L(self):
+        """(3) `numpy.ndarray` of `float`: The box lengths,
+        ``[Lx, Ly, Lz]``.
+
+        Can be set with a float which sets all lengths, or a length 3 vector.
+        """
+        return _to_three_array(self._cpp_obj.getL())
+
+    @L.setter
+    def L(self, new_L):
+        self._cpp_obj.setL(_make_scalar3(new_L))
+
+    @property
+    def Lx(self):
+        """float: The length of the box in the x dimension."""
+        return self.L[0]
+
+    @Lx.setter
+    def Lx(self, value):
+        L = self.L
+        L[0] = float(value)
+        self.L = L
+
+    @property
+    def Ly(self):
+        """float: The length of the box in the y dimension."""
+        return self.L[1]
+
+    @Ly.setter
+    def Ly(self, value):
+        L = self.L
+        L[1] = float(value)
+        self.L = L
+
+    @property
+    def Lz(self):
+        """float: The length of the box in the z dimension."""
+        return self.L[2]
+
+    @Lz.setter
+    def Lz(self, value):
+        L = self.L
+        L[2] = float(value)
+        self.L = L
+
+    # Box tilt based properties
+    @property
+    def tilts(self):
+        """(3) `numpy.ndarray` of `float`: The box tilts, ``[xy, xz, yz]``.
+
+        Can be set using one tilt for all axes or three tilts. If the box is 2D
+        ``xz`` and ``yz`` will automatically be set to zero."""
+        return np.array([self.xy, self.xz, self.yz])
+
+    @tilts.setter
+    def tilts(self, new_tilts):
+        new_tilts = _make_scalar3(new_tilts)
+        if self.is2D:
+            new_tilts.y = 0
+            new_tilts.z = 0
+        self._cpp_obj.setTiltFactors(new_tilts.x, new_tilts.y, new_tilts.z)
+
+    @property
+    def xy(self):
+        """float: The tilt for the xy plane."""
+        return self._cpp_obj.getTiltFactorXY()
+
+    @xy.setter
+    def xy(self, xy):
+        self.tilts = [xy, self.xz, self.yz]
+
+    @property
+    def xz(self):
+        """float: The tilt for the xz plane."""
+        return self._cpp_obj.getTiltFactorXZ()
+
+    @xz.setter
```

Comment:
What happens if you try to set xz with a 2D box?

Reply:
Good catch. We now raise an error.